src/Git/GitStatusListCtrl.cpp

```diff
@@ -661,8 +663,17 @@ void CGitStatusListCtrl::Show(unsigned int dwShow, unsigned int dwCheck /*=0*/,
 			else if (!UseStoredCheckStatus)
 			{
 				bool autoSelectSubmodules = !(entry->IsDirectory() && m_bDoNotAutoselectSubmodules);
-				if (((entry->m_Action & dwCheck) && !(m_bNoAutoselectMissing && entry->m_Action & CTGitPath::LOGACTIONS_MISSING) || dwShow & GITSLC_SHOWDIRECTFILES && m_setDirectFiles.find(path) != m_setDirectFiles.end()) && autoSelectSubmodules)
-					entry->m_Checked = true;
+				if (((entry->m_Action & dwCheck) &&
+					!(m_bNoAutoselectMissing && entry->m_Action & CTGitPath::LOGACTIONS_MISSING) ||
+					dwShow & GITSLC_SHOWDIRECTFILES && m_setDirectFiles.find(path) != m_setDirectFiles.end()
+					) && autoSelectSubmodules)
+				{
+					auto it = m_pathToChangelist.find(path);

```

Comment:
Would a new function `IsOnIgnoreonCommitChangeList`  result is more readable code?

Reply:
Maybe we can also get rid of `Do not automatically check submodules` in the future because of this feature.

Reply:
> 
> 
> Would a new function `IsOnIgnoreonCommitChangeList` result is more readable code?

I doubt it will be any better than direct code with a comment, this will be a function with just two lines of code and used only once (more spam in the header)

I'm not sure if this is any more readable but maybe replace internal `if` with:
```
// check if not in ignore-on-commit
entry->m_Checked = (it == m_pathToChangelist.end()) || (it->second.Compare(GITSLC_IGNORECHANGELIST) != 0); 

```